### PR TITLE
Prefer https over http

### PR DESCRIPTION
**Pull request by lencioni**
I checked all of the links in the JS style guide document and converted
as many as I could to use https.

### CODE REVIEW

**Comment by ljharb** [comment]
Presumably they've all been checked, since not every http link necessarily works as https?

**Comment by lencioni** [comment]
That's right. As I mentioned in my commit message, I checked them all and converted as many as I could. It wouldn't hurt to have a second check though!

File: README.md [hunk 1]
```diff
@@ -2846,25 +2846,25 @@ Other Style Guides
...
+  - [High Performance Web Sites: Essential Knowledge for Front-End Engineers](https://www.amazon.com/High-Performance-Web-Sites-Essential/dp/0596529309) - Steve Souders
+  - [Maintainable JavaScript](https://www.amazon.com/Maintainable-JavaScript-Nicholas-C-Zakas/dp/1449327680) - Nicholas C. Zakas
+  - [JavaScript Web Applications](https://www.amazon.com/JavaScript-Web-Applications-Alex-MacCaw/dp/144930351X) - Alex MacCaw
+  - [Pro JavaScript Techniques](https://www.amazon.com/Pro-JavaScript-Techniques-John-Resig/dp/1590597273) - John Resig
+  - [Smashing Node.js: JavaScript Everywhere](https://www.amazon.com/Smashing-Node-js-JavaScript-Everywhere-Magazine/dp/1119962595) - Guillermo Rauch
+  - [Secrets of the JavaScript Ninja](https://www.amazon.com/Secrets-JavaScript-Ninja-John-Resig/dp/193398869X) - John Resig and Bear Bibeault
   - [Human JavaScript](http://humanjavascript.com/) - Henrik Joreteg
-  - [Superhero.js](http://superherojs.com/) - Kim Joar Bekkelund, Mads Mobæk, & Olav Bjorkoy
+  - [Superhero.js](https://superherojs.com/) - Kim Joar Bekkelund, Mads Mobæk, & Olav Bjorkoy
```
Comment (lencioni) [inline README.md]: Damn, this one isn't right. I'll fix shortly.

Comment (lencioni) [inline README.md]: Fixed!

**Pull request merged**